### ISSUE TITLE
Architecture documentation for CAPZ networking

### ISSUE THREAD

**Issue by CecileRobertMichon** [kubernetes-sigs/cluster-api-provider-azure#1155]
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
As a new user of CAPZ, it would be good to get an overview of networking resources (eg. public IPs, Load Balancer, NICs, etc)  provisioned by CAPZ and how they relate to each other. 

We used to have https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/release-0.3/docs/design.md which was removed because it was outdated and was confusing folks more than it was helping.

Perhaps a diagram would be useful?

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by surajrana93** [comment]
@CecileRobertMichon I would like to help here. But I would require some guidance here as I am also a bit new on this

**Comment by CecileRobertMichon** [comment]
Sure, let me know when you are ready to start and maybe we can do a walkthough together

**Comment by CecileRobertMichon** [comment]
@surajrana93 are you still interested in helping with this? 

here's the link to the recorded walkthough: https://youtu.be/NekS355dq-U

**Comment by surajrana93** [comment]
@CecileRobertMichon  Unfortunately I am a bit busy these few weeks. You can go ahead and reassign. If I get sometime, I'll post an initial draft of the document and then you can assign it back.

**Comment by dmeytin** [comment]
Any progress on this issue?

**Comment by CecileRobertMichon** [comment]
No one is currently assigned to this @dmeytin, it's marked as help-wanted. I did a recorded walkthrough (link above) which should be mostly up to date. We did improve documentation on outbound connections (https://capz.sigs.k8s.io/topics/node-outbound-lb.html) and virtual networks (https://capz.sigs.k8s.io/topics/custom-vnet.html) since I first opened this issue but I still think having some sort of overview/visualization would still be valuable.
